probe/process/walker.go

```diff
@@ -76,3 +78,34 @@ var Walk = func(procRoot string, f func(*Process)) error {
 
 	return nil
 }
+
+var (
+	cache     []*Process
+	cacheLock sync.RWMutex
+)
```

Comment:
No; it would make the change a little bigger (would need a Walker interface etc) and I wasn't sure it was worth it.  But I can make that change if you like.